diff --git a/buildSrc/src/main/kotlin/com/velocitypowered/script/VelocityPublishPlugin.kt b/buildSrc/src/main/kotlin/com/velocitypowered/script/VelocityPublishPlugin.kt
index 8aa18b04..29c41ad0 100644
--- a/buildSrc/src/main/kotlin/com/velocitypowered/script/VelocityPublishPlugin.kt
+++ b/buildSrc/src/main/kotlin/com/velocitypowered/script/VelocityPublishPlugin.kt
@@ -33,9 +33,7 @@ import org.gradle.kotlin.dsl.getByType
 
 class VelocityPublishPlugin : Plugin<Project> {
     override fun apply(target: Project) = target.afterEvaluate {
-        if (target.name != "velocity-proxy") {
-            configure()
-        }
+        configure()
     }
     private fun Project.configure() {
         apply<JavaBasePlugin>()
@@ -45,10 +43,10 @@ class VelocityPublishPlugin : Plugin<Project> {
                 maven {
                     credentials(PasswordCredentials::class.java)
 
-                    name = "paper"
-                    val base = "https://papermc.io/repo/repository/maven"
-                    val releasesRepoUrl = "$base-releases/"
-                    val snapshotsRepoUrl = "$base-snapshots/"
+                    name = "diogotcRepository"
+                    val base = "https://repo.diogotc.com"
+                    val releasesRepoUrl = "$base/releases/"
+                    val snapshotsRepoUrl = "$base/snapshots/"
                     setUrl(if (version.toString().endsWith("SNAPSHOT")) snapshotsRepoUrl else releasesRepoUrl)
                 }
             }
