diff --git a/build-logic/src/main/kotlin/velocity-publish.gradle.kts b/build-logic/src/main/kotlin/velocity-publish.gradle.kts
index 51de5c67..1e85878a 100644
--- a/build-logic/src/main/kotlin/velocity-publish.gradle.kts
+++ b/build-logic/src/main/kotlin/velocity-publish.gradle.kts
@@ -8,10 +8,10 @@ extensions.configure<PublishingExtension> {
         maven {
             credentials(PasswordCredentials::class.java)
 
-            name = "paper"
-            val base = "https://repo.papermc.io/repository/maven"
-            val releasesRepoUrl = "$base-releases/"
-            val snapshotsRepoUrl = "$base-snapshots/"
+            name = "diogotcRepository"
+            val base = "https://repo.diogotc.com"
+            val releasesRepoUrl = "$base/releases/"
+            val snapshotsRepoUrl = "$base/snapshots/"
             setUrl(if (version.toString().endsWith("SNAPSHOT")) snapshotsRepoUrl else releasesRepoUrl)
         }
     }
diff --git a/proxy/build.gradle.kts b/proxy/build.gradle.kts
index 14eb4b19..3e743f36 100644
--- a/proxy/build.gradle.kts
+++ b/proxy/build.gradle.kts
@@ -4,6 +4,7 @@ plugins {
     application
     id("velocity-init-manifest")
     alias(libs.plugins.shadow)
+    id("velocity-publish")
 }
 
 application {
